yield_plugin/src/yield_plugin.cpp

```diff
@@ -372,11 +377,28 @@ namespace yield_plugin
     std::vector<carma_planning_msgs::msg::TrajectoryPlanPoint> jmt_trajectory_points;
     jmt_trajectory_points.push_back(original_tp.trajectory_points[0]);
 
-    double initial_time = 0;
-    double initial_accel = 0.0;
-    double goal_accel = 0.0;
-
-    RCLCPP_DEBUG_STREAM(nh_->get_logger(),"original_max_speed" << original_max_speed);
+    std::vector<double> original_traj_relative_downtracks = get_relative_downtracks(original_tp);
+    std::vector<double> calculated_speeds = {};
+    std::vector<double> new_relative_downtracks = {};
+    new_relative_downtracks.push_back(0.0);
+    calculated_speeds.push_back(initial_velocity);
+    const auto original_planning_duration = std::max(planning_time, get_trajectory_duration(original_tp));
+
+    // Calculating time_step to generate trajectory, which can be any constant.
+    // Only using original trajectory size here for potential performance improvement and dividing by 2 arbitrarily just to increase precision
+    const double average_time_step = std::max(0.1, original_planning_duration / original_tp.trajectory_points.size() / 2);
```

Comment:
time steps are determined by the speed because dist between points is fixed at 1 meter. Highest speed carma-platform is ever tested is 35mph, which is 15.64 m/s. So the time_step can be as small as 0.063sec, which maybe too small for our use case. I can specifically find the minimum time_step from the original (not average) so it is exactly the precision we need

Reply:
Yeah I like that approach. Let's push that to a different PR since that's could be a separate effort.